omegatrade-app/backend/app/models/company.model.js

```diff
@@ -0,0 +1,196 @@
+'user strict';
+const database = require('./../config/database.js');
+const {
+    v4: uuidv4
+} = require('uuid');
+var Company = function (company) {};
+
+Company.getAll = async function (cb) {
+    try {
+        const query = {
+            sql: 'select * from companies',
+        };
+        let result = await database.run(query);
+        if (result[0]) {
+            var rows = result[0].map((row) => row.toJSON());
+            cb(null, rows)
+        } else {
+            cb(null, null);
+        }
+    } catch (error) {
+        cb(error, null)
+    }
+}
+
+Company.getCompanyStock = async function (param, cb) {
+    try {
+        let companyId = param.companyId
+        let response = {}
+        const query = {
+            sql: 'select * from companies where companyId = @companyId',
+            params: {
+                companyId: companyId
+            }
+        };
+        let result = await database.run(query);
+        if (result[0]) {
+            var rows = result[0].map((row) => row.toJSON());
+            response.company = rows[0];
+        }
+        if (param.date) {
+            var whereClause = 'where companyId = @companyId and date > @date ORDER BY date';
+            var params = {
+                companyId: companyId,
+                date: param.date
+            };
+        } else {
+            var whereClause = 'where companyId = @companyId ORDER BY date';
+            var params = {
+                companyId: companyId
+            }
+        }
+        var query2 = {
+            sql: 'select date,currentValue from companyStocks ' + whereClause,
+            params: params
+        };
+        let stockResult = await database.run(query2);
+        if (stockResult[0]) {
+            var stockRows = stockResult[0].map((row) => row.toJSON());
+            response.stocks = stockRows;
+        }
+        cb(null, response)
+    } catch (error) {
+        cb(error, null)
+    }
+};
+
+Company.create = async function (params, result) {
```

Comment:
Why are there two different functions for `Company.create` and `Company.createData`? Are there circumstances where you actually want to create a `Company` without a name?